specifyweb/frontend/js_src/lib/components/Molecules/Dialog.tsx

```diff
@@ -167,6 +167,18 @@ export function Dialog({
 }): JSX.Element {
   const id = useId('modal');
 
+  const root = document.getElementById('root');
```

Comment:
By having this line in here, getElementById is executed on every render of every dialog - that would be wasteful in terms of performance - put this line outside the component